waspc/data/Generator/templates/sdk/wasp/server/operations/wrappers.ts

```diff
@@ -0,0 +1,234 @@
+ * @param userOperation The unauthenticated operation's definition.
+ * @param entities The unauthenticated operation's entity map .
+ * @returns The server-side API for the provided unauthenticated operation.
+ */
+export function createUnauthenticatedOperation<
+  OperationDefinition extends GenericOperationDefinition
+>(
+  userOperation: OperationDefinition,
+  entities: EntityMapFor<OperationDefinition>
+): UnauthenticatedOperationFor<OperationDefinition> {
+  async function operation(payload: Parameters<OperationDefinition>[0]) {
+    return userOperation(payload, {
+      entities,
+    })
+  }
+  // This cast is necessary because - When the Input is void, we want to present
+  // the function as not accepting a payload (which isn't consistent with how
+  // it's defined).
+  return operation as UnauthenticatedOperationFor<OperationDefinition>
+}
+
+{=# isAuthEnabled =}
+// PRIVATE API (used in SDK)
+// Explanation:
+// - Custom `_Awaited` and `_ReturnType` - Read the comments above their
+// definitions.
+// - `Parameters<OperationDefinition> extends []` - Same reason as described here:
+// https://github.com/wasp-lang/wasp/pull/1992/files#r1583040080
+/**
+ * Constructs the authenticated operation's server-side API type from its
+ * definition.
+ *
+ * @template OperationDefinition The type of the authenticated operation's
+ * definition.
+ */
+export type AuthenticatedOperationFor<
+  OperationDefinition extends GenericOperationDefinition
+> = Parameters<OperationDefinition> extends []
+  ? AuthenticatedOperation<void, _Awaited<_ReturnType<OperationDefinition>>>
+  : AuthenticatedOperation<
+      Parameters<OperationDefinition>[0],
+      _Awaited<_ReturnType<OperationDefinition>>
+    >
+
+/**
+ * The type of the context users must pass when calling an authenticated
+ * operation's server-side API.
+ */
+export type AuthenticatedOperationContext = { user: AuthUser }
+
+// PRIVATE API (used in SDK)
+/**
+ * Creates the server-side API for an authenticated operation.
+ *
+ * @template OperationDefinition The type of the authenticated operation's definition.
+ * @param userOperation The authenticated operation's definition.
+ * @param entities The authenticated operation's entity map .
+ * @returns The server-side API for the provided authenticated operation.
+ */
+export function createAuthenticatedOperation<
+  OperationDefinition extends GenericOperationDefinition
+>(
+  userOperation: OperationDefinition,
+  entities: EntityMapFor<OperationDefinition>
+): AuthenticatedOperationFor<OperationDefinition> {
+  async function operation(...args: AuthenticatedOperationArgsFor<OperationDefinition>) {
+    // To understand this function and its limitations, read
+    // https://github.com/wasp-lang/wasp/issues/2050
+    if (args.length < 1) {
+      // No arguments sent -> no user and no payload specified -> there's no way this was called correctly.
+      throw new Error('Invalid number of arguments')
```

Comment:
I'd maybe expand on the error and help the users to move forward. "You called an operation without arguments. Since this operation requires auth..."